### DISCUSSION TITLE
Buffer for Soapy Stream?

### DISCUSSION THREAD

**Discussion by bcshort** [cariboulabs/cariboulite#84] [Ideas]
Hi All,

I am wondering if anyone has some advise for having/increasing a network buffer for the Soapy interface?

I've found that anything less than wired ethernet from the pi to the remote network client introduces stuttering to the audio that I am guessing is network latency or wifi limitation issue.

I see some people here appear to have this working with a Pi Zero 2W - if you are connecting with Soapy over wifi network without issue, I am keen to know more :)

**Comment by konimaru** [comment]
For the record, installation max value is raised to 100MB (both recv/send). Default buffer setting during endpoint creation is 42MB. That's plenty I'd say. Option **remote:window** allows you to change it.

What format are you running with? **CS16** requires ~133Mbit, **CS8** ~65Mbit. Don't have a **zero** available so can't help you with hard evidence :)

**Comment by bcshort** [comment]
Thanks konimaru :)
Running at CS8 because the Pi3 does not appear to cope with CS16. I guess the other bear in the room might be the compute power of the Pi 3 vs a Pi 4 as well then. I'll do some investigation on the window size over the weekend and see what comes of it

**Comment by spgoblue** [comment]
I had a similar issue... it was all my fault.  I didn't notice that the power supply that I was using for the Raspberry Pi had too low of a voltage.  The Raspberry Pi had the Low Voltage warning and was therefore throttling (of course I ignored it at first).  A new power supply fixed the issue.  I also have to use Int8 as the sample type... the higher ones result in stuttering audio.  I have my Raspberry Pi (with Caribou Lite) and my SDR++ Client computer both on a wired network.
